plugins/modules/files/sapcar_extract.py

```diff
@@ -0,0 +1,220 @@
+    type: path
+  signature:
+    description:
+      - If true the Signature will be extracted
+    default: false
+    type: bool
+  manifest:
+    description:
+      - The Name of the manifest defaults to SIGNATURE.SMF.
+    default: "SIGNATURE.SMF"
+    type: str
+  remove:
+    description:
+      - If true the SAR/CAR file will be removed - !!!This should be used with caution!!!.
+    default: false
+    type: bool
+author:
+    - Rainer Leber (@RainerLeber)
+'''
+
+EXAMPLES = """
+# Extract SAR file - simple
+  - name: Run sapcar extract
+    community.general.sapcar_extract:
+      path: "~/source/hana.sar"
+    register: sapcar
+
+# Extract SAR file with destination
+  - name: Run sapcar extract
+    community.general.sapcar_extract:
+      path: "~/source/hana.sar"
+      dest: "~/test/"
+    register: sapcar
+
+# Extract SAR file with destination and download from webserver can be a fileshare as well
+  - name: Run sapcar extract
+    community.general.sapcar_extract:
+      path: "~/source/hana.sar"
+      dest: "~/dest/"
+      binary_path: "https://myserver/SAPCAR"
+    register: sapcar
+
+# Extract SAR file and delete SAR after extract
+  - name: Run sapcar extract
+    community.general.sapcar_extract:
+      path: "~/source/hana.sar"
+      remove: true
+    register: sapcar
+
+# Extract SAR file with manifest
+  - name: Run sapcar extract
+    community.general.sapcar_extract:
+      path: "~/source/hana.sar"
+      signature: true
+    register: sapcar
+
+# Extract SAR file with manifest and rename it
+  - name: Run sapcar extract
+    community.general.sapcar_extract:
+      path: "~/source/hana.sar"
+      manifest: "MyNewSignature.SMF"
+      signature: true
+    register: sapcar
+
+"""
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import open_url
+
+
+def getListOfFiles(dirName):
+    # create a list of file and directories
+    # names in the given directory
+    listOfFile = os.listdir(dirName)
+    allFiles = list()
+    # Iterate over all the entries
+    for entry in listOfFile:
+        # Create full path
+        fullPath = os.path.join(dirName, entry)
+        # If entry is a directory then get the list of files in this directory
+        if os.path.isdir(fullPath):
+            allFiles = allFiles + [fullPath]
+            allFiles = allFiles + getListOfFiles(fullPath)
+        else:
+            allFiles.append(fullPath)
+    return allFiles
+
+
+def downloadSAPCAR(binary_path):
+    bin_path = None
+    # download sapcar binary if url is provided otherwise path is returned
+    if binary_path is not None:
+        try:
+            with open_url(binary_path) as response:
+                with open("/tmp/sapcar", 'wb') as out_file:
+                    data = response.read()
+                    out_file.write(data)
+            os.chmod("/tmp/sapcar", 0o755)
```

Comment:
```suggestion
            os.chmod("/tmp/sapcar", 0o700)
```
I would stick to more strict access rights.